system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -278,6 +278,15 @@ this.TopStoriesFeed = class TopStoriesFeed {
           this.init();
         }
         break;
+      case at.PLACES_LINK_BLOCKED:
+        if (this.stories) {
+          this.stories = this.stories.filter(s => s.url !== action.data.url);
+        }
+        if (this.spocs) {
+          this.spocs = this.spocs.filter(s => s.url !== action.data.url);
+        }
+        SectionsManager.updateSection(SECTION_ID, {rows: this.stories}, true);
```

Comment:
Would be nice to check if the `blocked_url` has actually affected `this.stories`. If so, then updates the section, otherwise, nothing else needs to be done.

Reply:
To be clear, I assume the `PLACES_URL_BLOCKED` could be triggered from other sections other than Topstories, like dismissing a link from Topsites or Highlights.